pkg/controller/controller_binding.go

```diff
@@ -261,6 +263,56 @@ func (c *controller) reconcileServiceInstanceCredential(binding *v1alpha1.Servic
 			return nil
 		}
 
+		parametersChecksum := ""
+		if parameters != nil {
+			parametersChecksum, err = generateChecksumOfParameters(parameters)
+			if err != nil {
+				s := fmt.Sprintf(`Failed to generate the parameters checksum to store in the Status of ServiceInstanceCredential "%s/%s": %s`, binding.Namespace, binding.Name, err)
+				glog.Info(s)
+				c.recorder.Eventf(binding, api.EventTypeWarning, errorWithParameters, s)
+				c.setServiceInstanceCredentialCondition(
+					toUpdate,
+					v1alpha1.ServiceInstanceCredentialConditionReady,
+					v1alpha1.ConditionFalse,
+					errorWithParameters,
+					s)
+				if _, err := c.updateServiceInstanceCredentialStatus(toUpdate); err != nil {
+					return err
+				}
+				return err
+			}
+		}
+
+		var rawParametersWithRedaction *runtime.RawExtension
+		if parametersWithSecretsRedacted != nil {
```

Comment:
seems like this can be part of the above if statement.

We should validate that the parameters and redactedParameters are either both not-nil or both nil.